eth/stagedsync/stage_snapshots.go

```diff
@@ -334,7 +334,11 @@ func FillDBFromSnapshots(logPrefix string, ctx context.Context, tx kv.RwTx, dirs
 			if err := blockReader.HeadersRange(ctx, func(header *types.Header) error {
 				blockNum, blockHash := header.Number.Uint64(), header.Hash()
 				td.Add(td, header.Difficulty)
-
+				// What can happen if chaindata is deleted is that maybe header.seg progress is lower or higher than
```

Comment:
i think this logic must be inside HeadersRange func

Reply:
and from this perspective: maybe need re-visit other blockReader methods and add similar condition there also